spec/ReactTags.spec.js

```diff
@@ -453,7 +454,18 @@ describe('React Tags', () => {
       // As of JSDom 9.10.0 scrollWidth is a getter only and always 0
       // TODO: can we test this another way?
       expect(input.style.width).toBeTruthy()
-      expect(window.getComputedStyle(input).width).toEqual(sizer.scrollWidth + 2 + 'px')
+      expect(window.getComputedStyle(input).width).toEqual(sizer.scrollWidth + 20 + 'px')
+    })
+
+    it('resizes input to match random sizer width', () => {
+      const input = $('input')
+      let rand
+      spyOn(Input.prototype, 'getScrollWidth').and.callFake(() => { rand = Math.floor(Math.random() * Math.floor(100)); return rand })  // scrollWidth was always 0 in test before
```

Comment:
I like this idea, much better than `0`!

Reply:
Although... this will actually create a problem itself, because the current `updateInputWidth()` method will only call `setState()` if the calculated width changes. With this it will change every time and likely create a loop.

Reply:
First of all, thank you for your answer. We created this test because we wanted to make sure our temporary solution fixes this infinite loop issue. The purpose of the test is to create a long loop to simulate the situation when we add many long tags after each other like we did in our application. This was our best idea to reliably recreate this issue. We know according to the current implementation it's not likely a long loop like this occurs many times, but it's also not an edge case -- we seen the error to happen more than a few times in our production code. In our opinion our solution is far from good, but this can be a serious issue and we wanted to talk about possible solutions and cases with you. Again, thank you for your time and effort on this topic.

Reply:
I'm happy to work with you to help solve the problem and it's reasonable to suspect the resize logic as it may potentially create an open loop.

I also really appreciate your efforts creating a test case - not many people do - but I think we'll need to try and create a real, isolated test case because creating a long loop on purpose then adding code to then avoid it lacks context and I worry that is potentially hiding the true problem.

Reply:
Thank you! It's a good idea to write an isolated test but I am not sure how to recreate the error without a long loop. Do you have any idea how we should write this new test to better simulate the issue? Or if you already wrote it please keep me updated, I would be really happy to help if I can.